### PR TITLE
Fl/connection pool

### PR DESCRIPTION
**Pull request by lemaitre-aneo**
Use a connection pool and remove the mutex in order to fix some "Go Away" errors while improving performance.

### CODE REVIEW

**Comment by lemaitre-aneo** [comment]
This continues to fix the GoAway issue. Preliminary tests show a 20% improvements on performance compared to the mutex.

**Review by ddubuc — COMMENTED** [review]
Except Log verbosity and function renaming, it's ok

File: Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/ChannelPool.cs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  {
+    if (pool_.TryTake(out var channel))
+    {
+      logger_?.LogDebug("Acquired already existing channel {channel} from pool",
+                        channel);
+      return channel;
+    }
+
+    channel = channelFactory_();
+    logger_?.LogInformation("Created and acquired new channel {channel} from pool",
```
[the remark this answers is not in the document]
Comment (lemaitre-aneo) [reply Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/ChannelPool.cs:75]: Those events should be really rare as they happen only when the pool is empty. I think it is useful to have the log at the information level.

File: Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/BaseClientSubmitter.cs [hunk 2]
```diff
@@ -60,17 +60,18 @@ public class BaseClientSubmitter<T>
...
                              [CanBeNull] ILoggerFactory loggerFactory   = null,
                              int                        chunkSubmitSize = 500)
   {
-    Logger           = loggerFactory?.CreateLogger<T>();
-    TaskService      = new Tasks.TasksClient(channelBase);
-    ResultService    = new Results.ResultsClient(channelBase);
-    SubmitterService = new Api.gRPC.V1.Submitter.Submitter.SubmitterClient(channelBase);
+    channelPool_ = channelPool;
+    Logger       = loggerFactory?.CreateLogger<T>();
+    //TaskService      = new Tasks.TasksClient(channel);
```
Comment (aneojgurhem) [inline Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/BaseClientSubmitter.cs:72]: On peut enlever ces commentaires

File: Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/ClientServiceConnector.cs [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
-using System.Runtime.InteropServices;
-using System.Security.Cryptography.X509Certificates;
-#else
-using Grpc.Core;
-#endif
 using System;
 using System.IO;
 using System.Net.Http;
 
+using Grpc.Core;
```
Comment (aneojgurhem) [inline Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/ClientServiceConnector.cs:5]: Il était dans le else du if > net5, il devrait le rester non ?

File: Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/ClientServiceConnector.cs [hunk 4]
```diff
@@ -42,17 +41,17 @@ public class ClientServiceConnector
   /// <summary>
   ///   Open Connection with the control plane with mTLS authentication
   /// </summary>
-  /// <param name="endPoint"></param>
+  /// <param name="endPoint">The address and port of control plane</param>
```
Comment (aneojgurhem) [inline Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/ClientServiceConnector.cs:44]: 💯

File: Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/ClientServiceConnector.cs [hunk 5]
```diff
@@ -168,4 +167,59 @@ public class ClientServiceConnector
...
   }
+
+  /// <summary>
+  ///   Create a connection pool to the control plane with or without SSL and no mTLS
+  /// </summary>
+  /// <param name="endPoint">The address and port of control plane</param>
+  /// <param name="sslValidation">Optional : Check if the ssl must have a strong validation (default true)</param>
+  /// <param name="loggerFactory">Optional : the logger factory to create the logger</param>
+  /// <returns></returns>
+  public static ChannelPool ControlPlaneConnectionPool(string         endPoint,
```
[the remark this answers is not in the document]
Comment (aneojgurhem) [reply Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/ClientServiceConnector.cs:178]: That's not really exposed externally so it does not really matter. Having it internal would have been better

File: Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/ClientServiceConnector.cs (hunk 3 above)
Comment (lemaitre-aneo) [reply Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/ClientServiceConnector.cs:5]: `ChannelBase` vient de `Grpc.Core`.

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**